core/services/workflows/queue.go

```diff
@@ -0,0 +1,53 @@
+package workflows
+
+import (
+	"context"
+	"sync"
+)
+
+type stepRequest struct {
+	stepRef string
+	state   executionState
+}
+
+type queue[T any] struct {
```

Comment:
How is this queue different than just a channel with enough capacity to basically be considered "infinite unless a catastrophic overload happens"?

Reply:
Good point :) I've removed -- my thinking was that we could use this to peek from the channel, but we don't need this so functionality in the end, so going to remove it.